lib/std/builtin.zig

```diff
@@ -47,25 +47,25 @@ pub const subsystem: ?SubSystem = blk: {
     }
 };
 
-/// This data structure is used by the Zig language code generation and
-/// therefore must be kept in sync with the compiler implementation.
 pub const StackTrace = struct {
+    // This data structure is used by the Zig language code generation and
+    // therefore must be kept in sync with the compiler implementation.
```

Comment:
FWIW personally I'd set the comments to non-documenting comments as you did, but I'd leave it above the struct definition. Otherwise, it appears as though it's documenting the `index` member.

Reply:
My thinking was that at some point there will be actual doc comments for these. The options are

```
/// description of how to use StackTrace
pub const StackTrace = struct {
// comment for maintainers of StackTrace
```

and

```
// comment for maintainers of StackTrace
/// description of how to use StackTrace
pub const StackTrace = struct {
```

I thought the first made more sense.

Reply:
Hmm, I see your point. I was actually thinking about a cross between the two, something like:

```zig
/// docs
// comments for maintainers
pub const StackTrace = struct {
```

Would that make more sense?

Reply:
It would make more sense than the second option I had up above, but I don't see a big advantage over putting the comment inside the scope to which it refers. Honestly, I didn't consider it because I didn't think it would work, given the documentation about doc comments, but it does.

The whole thing seems kinda bike sheddy to me though. If you feel strongly, you're welcome to amend the PR. I'd kinda like to get this merged or rejected before another major API change forces me to either abandon it or do it over again.

Reply:
```
/// docs
// comments for maintainers
```
According to the documentation, this pattern will eventually cause a compile error.
>  Doc comments are only allowed in certain places; eventually, it will become a compile error to have a doc comment in an unexpected place, such as in the middle of an expression, **or just before a non-doc comment.**

Reply:
Good catch @justjosias! FWIW, I'm happy to leave it as-is @nmichaels.